cmd/util/cmd/rollback-executed-height/cmd/rollback-executed-height.go

```diff
@@ -0,0 +1,206 @@
+func removeExecutionResultsFromHeight(
+	protoState protocol.State,
+	transactionResults storage.TransactionResults,
+	commits storage.Commits,
+	chunkDataPacks storage.ChunkDataPacks,
+	results storage.ExecutionResults,
+	fromHeight uint64) error {
+	log.Info().Msgf("removing results for blocks from height: %v", flagHeight)
+
+	root, err := protoState.Params().Root()
+	if err != nil {
+		return fmt.Errorf("could not get root: %w", err)
+	}
+
+	if fromHeight <= root.Height {
+		return fmt.Errorf("can only remove results for block above root block. fromHeight: %v, rootHeight: %v", fromHeight, root.Height)
+	}
+
+	final, err := protoState.Final().Head()
+	if err != nil {
+		return fmt.Errorf("could get not finalized height: %w", err)
+	}
+
+	if fromHeight > final.Height {
+		return fmt.Errorf("could not remove results for unfinalized height: %v, finalized height: %v", fromHeight, final.Height)
+	}
+
+	finalRemoved := 0
+	total := int(final.Height-fromHeight) + 1
+
+	// removing for finalized blocks
+	for height := fromHeight; height <= final.Height; height++ {
+		head, err := protoState.AtHeight(height).Head()
+		if err != nil {
+			return fmt.Errorf("could not get header at height: %w", err)
+		}
+
+		blockID := head.ID()
+
+		err = removeForBlockID(commits, transactionResults, results, chunkDataPacks, blockID)
+		if err != nil {
+			return fmt.Errorf("could not remove result for finalized block: %v, %w", blockID, err)
+		}
+
+		finalRemoved++
+		log.Info().Msgf("result at height :%v has been removed. progress (%v/%v)", height, finalRemoved, total)
+	}
+
+	// removing for pending blocks
+	pendings, err := protoState.Final().Descendants()
+	if err != nil {
+		return fmt.Errorf("could not get pending block: %w", err)
+	}
+
+	pendingRemoved := 0
+	total = len(pendings)
+
+	for _, pending := range pendings {
+		err = removeForBlockID(commits, transactionResults, results, chunkDataPacks, pending)
+
+		if err != nil {
+			return fmt.Errorf("could not remove result for pending block: %v, %w", pending, err)
+		}
+
+		pendingRemoved++
+		log.Info().Msgf("result for pending block :%v has been removed. progress (%v/%v) ", pending, pendingRemoved, total)
+	}
+
+	log.Info().Msgf("removed height from %v. removed for %v finalized blocks, and %v pending blocks",
+		fromHeight, finalRemoved, pendingRemoved)
+
+	return nil
+}
+
+// removeForBlockID remove block execution related data for a given block.
+func removeForBlockID(
+	commits storage.Commits,
+	transactionResults storage.TransactionResults,
+	results storage.ExecutionResults,
+	chunks storage.ChunkDataPacks,
+	blockID flow.Identifier) error {
+	result, err := results.ByBlockID(blockID)
+	if errors.Is(err, storage.ErrNotFound) {
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("could not find result for block %v: %w", blockID, err)
+	}
+
+	// remove chunks
+	for _, chunk := range result.Chunks {
+		chunkID := chunk.ID()
+		err := chunks.Remove(chunkID)
+		if errors.Is(err, storage.ErrNotFound) {
```

Comment:
Would be good to log here - I don't think there is an expected path where we have an ER but not the chunks (or only some of the chunks).